governator-core/src/main/java/com/netflix/governator/ScanningModuleBuilder.java

```diff
@@ -0,0 +1,231 @@
+
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+/**
+ * When installed this module will scan for annotated classes and creating appropriate bindings.  
+ * The specific annotation to scan and binding semantics are captured in a {@link AnnotatedClassScanner}.
+ * 
+ * The following example shows how to install a module that creates bindings for classes containing
+ * the AutoBindSingleton annotation.
+ * 
+ * <pre>
+ * {@code 
+ * install(new ScanningModuleBuilder().
+ *      .forPackages("org.example")
+ *      .scanForAnnotatedClasses(new AutoBindSingletonAnnotatedClassScanner())
+ *      .build()
+ * }
+ * </pre>
+ */
+public class ScanningModuleBuilder {
+    private List<String> packages = new ArrayList<>();
+    private List<AnnotatedClassScanner> scanners = new ArrayList<>();
+    private ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+    private Predicate<String> excludeRule = Predicates.alwaysFalse();
+    
+    /**
+     * Specify a custom class loader to use.  If not specified Thread.currentThread().getContextClassLoader()
+     * will be used.
+     * 
+     * @param classLoader
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder usingClassLoader(ClassLoader classLoader) {
+        this.classLoader = classLoader;
+        return this;
+    }
+    
+    /**
+     * Set of packages to scan.  
+     * 
+     * @param packages
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder forPackages(String... packages) {
+        return forPackages(Arrays.asList(packages));
+    }
+    
+    /**
+     * Set of packages to scan.  
+     * 
+     * @param packages
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder forPackages(Collection<String> packages) {
+        this.packages = new ArrayList<>(packages);
+        return this;
+    }
+    
+    /**
+     * Specify an {@link AnnotatedClassScanner} to process classes from the set of specified packages.  
+     * 
+     * @param packages
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder scanForAnnotatedClasses(AnnotatedClassScanner scanner) {
+        scanners.add(scanner);
+        return this;
+    }
+    
+    /**
+     * Exclude bindings for any class matching the specified predicate.
+     * @param classes
+     * @return Builder for chaining
+     */
+    public ScanningModuleBuilder excludeClasses(Predicate<String> predicate) {
```

Comment:
should be clearer what the predicate is receiving.  Is it the fully-qualified classname of the candidate to scan?